sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -48,7 +48,7 @@ class InMemoryCatalog extends ExternalCatalog {
   private val catalog = new scala.collection.mutable.HashMap[String, DatabaseDesc]
 
   private def filterPattern(names: Seq[String], pattern: String): Seq[String] = {
-    val regex = pattern.replaceAll("\\*", ".*").r
+    val regex = pattern.replaceAll("\\.\\*", "\\*").replaceAll("\\*", "\\.\\*").r
```

Comment:
Regarding this change, I think it depends on what is the wildcard for any character(s). If it is `*` (following Hive), then, we should keep the original line.